src/host/path_normalize.c

```diff
@@ -70,8 +70,8 @@ static void* normalize_substring(const char* str, const char* endPtr, char* writ
 			}
 		}
 
-		/* add to the result, filtering out duplicate slashes */
-		if (ch != '/' || last != '/') {
+		/* add to the result, filtering out duplicate slashes, except when they are leading slashes */
+		if (str == &source[1] || (ch != '/' || last != '/')) {
```

Comment:
I think this is the line I was meant to comment on. Is `source` guaranteed to not be an empty string here?

Reply:
Without testing, I believe so. if the string is empty, then `str` will be equal to `endPtr` and the loop condition `while (str != endPtr)` will fail.

Reply:
After some testing I found out that it never attempt to normalize an empty string, thanks to https://github.com/premake/premake-core/blob/a713eccf96e352568b1798c71d14d4c5221a4a73/src/host/path_normalize.c#L108